### PR TITLE
Feature 240/testcontainer agenda

### PR DESCRIPTION
**Pull request by Maroju100**
- Agenda item tests converted to test containers
- Ran IntelliJ coverage on newly created controller tests
- Code coverage of 80% for involved classes ensured

### CODE REVIEW

File: server/src/test/java/com/objectcomputing/checkins/services/agenda_item/AgendaItemControllerTest.java [hunk 1]
```diff
@@ -99,42 +94,83 @@ void testCreateANullAgendaItem() {
         assertEquals(request.getPath(), href.asText());
         assertEquals(HttpStatus.BAD_REQUEST, responseException.getStatus());
 
-        verify(agendaItemServices, never()).save(any(AgendaItem.class));
     }
 
     @Test
-    void testLoadAgendaItems() {
+    void testCreateAnAgendaItemForExistingCheckInId() {
```
Comment (Bagurp) [inline server/src/test/java/com/objectcomputing/checkins/services/agenda_item/AgendaItemControllerTest.java:100]: should this be non-existing checkinId?

File: server/src/test/java/com/objectcomputing/checkins/services/agenda_item/AgendaItemControllerTest.java [hunk 2]
```diff
@@ -99,42 +94,83 @@ void testCreateANullAgendaItem() {
...
+        String error = Objects.requireNonNull(body).get("message").asText();
+        String href = Objects.requireNonNull(body).get("_links").get("self").get("href").asText();
+
+        assertEquals(request.getPath(),href);
+        assertEquals(String.format("CheckIn %s doesn't exist",agendaItemCreateDTO.getCheckinid()),error);
+
+    }
+
+    @Test
+    void testCreateAnAgendaItemForExistingMemberIdId() {
```
Comment (Bagurp) [inline server/src/test/java/com/objectcomputing/checkins/services/agenda_item/AgendaItemControllerTest.java:121]: same here?

**Comment by bhatiam12** [comment]
@Maroju100 is this being used anywhere ?
```
   public Set<AgendaItem> readAll() {
        Set<AgendaItem> agendaItems = new HashSet<>();
        agendaItemRepo.findAll().forEach(agendaItems::add);
        return agendaItems;
    }```

If not please remove it as we don't have tests to cover this.

File: server/src/test/java/com/objectcomputing/checkins/services/agenda_item/AgendaItemControllerTest.java (hunk 2 above)
Comment (Maroju100) [reply server/src/test/java/com/objectcomputing/checkins/services/agenda_item/AgendaItemControllerTest.java]: updated

File: server/src/test/java/com/objectcomputing/checkins/services/agenda_item/AgendaItemControllerTest.java (hunk 1 above)
Comment (Maroju100) [reply server/src/test/java/com/objectcomputing/checkins/services/agenda_item/AgendaItemControllerTest.java]: yes, updated

**Comment by Maroju100** [comment]
> @Maroju100 is this being used anywhere ?
> 
> ```
>    public Set<AgendaItem> readAll() {
>         Set<AgendaItem> agendaItems = new HashSet<>();
>         agendaItemRepo.findAll().forEach(agendaItems::add);
>         return agendaItems;
>     }```
> 
> If not please remove it as we don't have tests to cover this.
> ```

Should we keep this, many services have defined this function but don't have tests for them.

**Comment by Maroju100** [comment]
> > @Maroju100 is this being used anywhere ?
> > ```
> >    public Set<AgendaItem> readAll() {
> >         Set<AgendaItem> agendaItems = new HashSet<>();
> >         agendaItemRepo.findAll().forEach(agendaItems::add);
> >         return agendaItems;
> >     }```
> > 
> > If not please remove it as we don't have tests to cover this.
> > ```
> 
> Should we keep this, many services have defined this function but don't have tests for them.
It looks like readall() is part of definition, we can keep this
https://github.com/oci-labs/check-ins/issues/140

File: server/src/test/java/com/objectcomputing/checkins/services/agenda_item/AgendaItemControllerTest.java [hunk 3]
```diff
@@ -10,63 +16,56 @@
...
-        AgendaItemCreateDTO agendaItemCreateDTO = new AgendaItemCreateDTO();
-        agendaItemCreateDTO.setCheckinid(UUID.randomUUID());
-        agendaItemCreateDTO.setCreatedbyid(UUID.randomUUID());
-        agendaItemCreateDTO.setDescription("dnc");
 
-        AgendaItem a = new AgendaItem(agendaItemCreateDTO.getCheckinid(), agendaItemCreateDTO.getCreatedbyid(), agendaItemCreateDTO.getDescription());
+        MemberProfile memberProfile = createADefaultMemberProfile();
+        MemberProfile memberProfileForPDL = createADefaultMemberProfileForPdl(memberProfile);
+
+        CheckIn checkIn = createADefaultCheckIn(memberProfile,memberProfileForPDL);
```
Comment (bhatiam12) [inline server/src/test/java/com/objectcomputing/checkins/services/agenda_item/AgendaItemControllerTest.java:45]: `createADeafultAgendaItem()` expects a checkin and memberprofile so this need to be changed.

File: server/src/test/java/com/objectcomputing/checkins/services/agenda_item/AgendaItemControllerTest.java [hunk 4]
```diff
@@ -99,42 +94,83 @@ void testCreateANullAgendaItem() {
...
+        assertEquals(String.format("CheckIn %s doesn't exist",agendaItemCreateDTO.getCheckinid()),error);
+
+    }
+
+    @Test
+    void testCreateAnAgendaItemForNonExistingMemberIdId() {
+        MemberProfile memberProfile = createADefaultMemberProfile();
+        MemberProfile memberProfileForPDL = createADefaultMemberProfileForPdl(memberProfile);
+
+        CheckIn checkIn = createADefaultCheckIn(memberProfile,memberProfileForPDL);
```
Comment (bhatiam12) [inline server/src/test/java/com/objectcomputing/checkins/services/agenda_item/AgendaItemControllerTest.java:125]: same here. need to pass checkin and member profile.

**Pull request merged**